src/runtime/cuda/cuda_device_api.cc

```diff
@@ -297,8 +298,13 @@ class CUDADeviceAPI final : public DeviceAPI {
 
 typedef dmlc::ThreadLocalStore<CUDAThreadEntry> CUDAThreadStore;
 
+// TODO(cliu): cuda streams should depend on the current device, therefore we should set device
+//             before setting stream.
 CUDAThreadEntry::CUDAThreadEntry()
     : pool(kDLGPU, CUDADeviceAPI::Global()) {
+  TensorDispatcher* td = TensorDispatcher::Global();
+  if (td->IsAvailable())
+    stream = td->CUDAGetCurrentStream();
 }
```

Comment:
Could this be error-prone of not setting device here? Now, dgl internal cuda stream is stored as a thread-local variable in CUDAThreadEntry. I believe it is not exposed/related to the current device id in use (DGLContext), right?

Now, I guess it relies on whether the device has been appropriately set at beginning by pytorch.

Reply:
> the device has been appropriately set at beginning by pytorch.

I think it should be fine. For the single GPU case, PyTorch will use device 0 by default. While for multi-GPU DDP, users are required to set the device for each process.